### ISSUE TITLE
Playing sound hangs the VM in Safari

### ISSUE THREAD

**Issue by dpinela** [copy/v86#286]
**How to reproduce**: Open the Windows 98 demo linked in the README. Open the WAV file in the desktop, or, alternatively, try to play a sound from the Sound control panel.

**What should happen**: Sound plays.

**What happens**: The VM hangs. The following error comes up in the console:

```
TypeError: e.copyToChannel is not a function. (In 'e.copyToChannel(a[0],0)', 'e.copyToChannel' is undefined)
ug — speaker.js:965
(anonymous function) — speaker.js:903
send — bus.js:77
Rd — sb16.js:1821
(anonymous function) — sb16.js:1741
(anonymous function) — dma.js:355
set — lib.js:119
Pe — dma.js:344
Sd — sb16.js:1747
(anonymous function) — sb16.js:232
send — bus.js:77
Gc — speaker.js:1009
(anonymous function) — speaker.js:909
send — bus.js:77
Fb — sb16.js:1737
Tc — dma.js:258
xh — dma.js:202
[native code]
cb — io.js:343
(anonymous function) — instructions.js:1211
we — cpu.js:1182
Ca — cpu.js:1113
ra — cpu.js:1099
jb — cpu.js:1085
(anonymous function) — main.js:108
```

The problem line in speaker.js:

```js
        buffer.copyToChannel(data[0], 0);
```

Some research indicates that the problem is that Safari doesn't support the [copyToChannel][] method on AudioBuffer.

**Browser**: Safari 12.1
**OS**: macOS 10.14.4

[copyToChannel]: https://stackoverflow.com/questions/32640573/alternative-to-audiocontext-copytochannel-in-safari-and-edge#comment91867776_32640573

**Comment by ErnWong** [comment]
Thanks for reporting this! I'll look into getting sound working on safari. There seem to be other problems in addition to the one you found as well.

**Comment by ErnWong** [comment]
Here's a [PR](https://github.com/copy/v86/pull/287) that fixes the `copyToChannel` bug.

I still couldn't get sound to work on my friend's iPad, but I couldn't get some kind of developer tools to connect to it yet to see what the problem is.

Hey @dpinela mind if I ask you to try out [this PR](https://github.com/copy/v86/pull/287) and see if the sound file plays alright on your macOS?

**Comment by dpinela** [comment]
Now it no longer crashes, and it does try to play sound, but it's *extremely* choppy, to the point where most of the time you just don't hear anything.

Also, after playing sound (or trying to), it consistently gets into a state where some animations only update when you move the mouse, like opening a menu or minimising a window. This last part happens even in Chrome, where sound otherwise works fine (I wonder if it's actually a separate issue?)

**Comment by ErnWong** [comment]
>...but it's extremely choppy...

Ah I see. I'll try and get hold of a mac to work on it one day, but that day probably won't be soon. (Pull requests from anyone are welcome!)

> ...some animations only update when you move the mouse...
> ...(I wonder if it's actually a separate issue?)

I originally thought it was a separate issue unrelated to audio, but I checked just now and it only enters this state after it starts playing audio, and it doesn't happen when I open Windows Media Player with v86's audio device disabled.

Well, it could also be a problem unrelated to audio, but which, wasn't revealed until audio was implemented and could be played by Windows Media Player. I have vague memories of seeing this behaviour outside of Windows Media Player when running v86 in debug mode, so it might be triggered by slow performance instead. (But this was from a year ago and I can't remember exactly if that was the case).

But whatever's causing it, yes I'll file this as a separate issue. Thanks for your help @dpinela !
